File: EStrutur_DATOS_TAREA02/JAVA Estudiante/Estudiante.java


class Estudiante{

    private String nombre;
    private int id;
    private int edad;
    private double promedio;

    public Estudiante(String nombre,int id, int edad, double promedio){
        this.nombre = nombre;
        this.edad = edad;
        this.promedio = promedio;
        this.id = id;
    }

    public int getEdad() {
        return edad;
    }

    public String getNombre() {
        return nombre;
    }

    public double getPromedio() {
        return promedio;
    }

    public int getId() {
        return id;
    }

    public void setEdad(int edad) {
        this.edad = edad;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public void setPromedio(double promedio) {
        this.promedio = promedio;
    }

    @Override
    public String toString(){
        return "Nombre: "+ nombre + " |ID: "+ id + " |Edad: " + edad + " |Promedio: " + promedio; 
    }
}

File: Estructura_Datos_Tarea03/C++Estudiante/Estudiante.cpp
#include <iostream>
#include <string>
using namespace std;

class Estudiante{
    protected:

        string nombre;
        int edad;
        int id;
        double promedio;

    public:
    Estudiante(string nombre, int edad, int id, double promedio){
        this->nombre = nombre;
        this->edad = edad;
        this-> id = id;
        this-> promedio = promedio;
    }

    int getEdad(){
        return edad;
    }

    string getNombre(){
        return nombre;
    }

    double getPromedio(){
        return promedio;
    }

    int getId(){
        return id;
    }

    void setEdad(int nuevaEdad){
        edad = nuevaEdad;
    }

    void setNombre(string nuevoNombre){
        nombre = nuevoNombre;
    }

    void setPromedio(double nuevoPromedio){
        promedio = nuevoPromedio;
    }

    void mostrarInfoEstudiante(){
        cout << "- " << "Nombre: "<< nombre << " |ID: "<<id
            <<" |Edad: "<<edad<< " |Promedio: " <<promedio << endl;
    }
};